content/lessons/chapter-6/put-it-together-6.tsx

```diff
@@ -1,34 +1,234 @@
+
+# Explained in step 6
+scriptcode = "1976a914" + pubkey_hash + "88ac"
+
+tx = Transaction()
+# YOUR CODE HERE
+`,
+    validate: async (answer: string) => {
+      if (
+        answer.slice(0, 248) !==
+        '020000000001018e74531c4516169a7cc84d3f65c216a39dcb24cae59d1fd76e6320c93116088a0100000000ffffffff0200e1f50500000000220020422e079e04cdec4dd15ccf0b3fd0c742eea8b067bf06c2b489c6efd05abf1fd19851c82000000000160014b234aee5ee74d7615c075b4fe81fd8ace54137f202'
+      ) {
+        return [false, 'Nope! Try again.']
+      }
+      if (answer.slice(250, 252) !== '30') {
+        return [false, 'Nope! Try again.']
+      }
+      const size = (parseInt(answer.slice(248, 250), 16) - 1) * 2
+      if (answer.slice(250 + size, 250 + size + 4) !== '0121') {
+        return [false, 'Nope! Try again.']
+      }
+      if (answer.slice(-8) !== '00000000') {
+        return [false, 'Nope! Try again.']
+      }
+      return [true, 'Nicely Done']
+    },
+    constraints: [
+      {
+        range: [
+          countLines(combinedCode) + 16,
+          1,
+          countLines(combinedCode) + 17,
+          1,
+        ],
+        allowMultiline: true,
+      },
+    ],
+    hiddenRange: [1, 1, countLines(combinedCode), 1],
+  }
+
+  const config: EditorConfig = {
+    defaultLanguage: 'javascript',
+    languages: {
+      javascript,
+      python,
+    },
+  }
 
+  const [language, setLanguage] = useState(config.defaultLanguage)
+  const handleSelectLanguage = (language: string) => {
+    setLanguage(language)
+  }
   return (
-    <Introduction lang={lang} imagePosition="object-[50%_27%]">
-      <Text className="text-lg md:text-xl">
-        {t('chapter_five.intro_one.paragraph_one')}
-      </Text>
-      <Text className="mt-4 text-lg md:text-xl">
-        {t('chapter_five.intro_one.paragraph_two')}
-      </Text>
-      <Text className="mt-4 text-lg md:text-xl">
-        {t('chapter_five.intro_one.paragraph_three')}
-      </Text>
-      <Text className="mt-4 text-lg md:text-xl">
-        {t('chapter_five.intro_one.paragraph_four')}
-      </Text>
-      <Text className="mt-4 text-lg md:text-xl">
-        {t('chapter_five.intro_one.paragraph_five')}
-      </Text>
-    </Introduction>
+    !isLoading &&
+    combinedCode && (
+      <ScriptingChallenge
+        lang={lang}
+        config={config}
+        saveData
+        lessonKey={getLessonKey('chapter-6', 'put-it-together-6')}
+        successMessage={t('chapter_six.put_it_together_six.success')}
+        onSelectLanguage={handleSelectLanguage}
+        loadingSavedCode={isLoading}
+      >
+        <LessonInfo>
+          <Text className="font-nunito text-xl text-white">
+            {t('chapter_six.put_it_together_six.paragraph_one')}
+          </Text>
+          <Text className="mt-4 font-nunito text-xl text-white">
+            {t(`chapter_six.put_it_together_six.paragraph_two`)}
+          </Text>
+          <ul className="list-inside list-disc font-nunito text-white">
```

Comment:
The first bullet needs to have a new word wrap schema and I think we should use list-outside... I will get back to this one to check how other pages are